compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1483,13 +1483,24 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
                 .emit();
             }
             FulfillmentErrorCode::CodeConstEquateError(ref expected_found, ref err) => {
-                self.report_mismatched_consts(
+                let mut diag = self.report_mismatched_consts(
                     &error.obligation.cause,
                     expected_found.expected,
                     expected_found.found,
                     err.clone(),
-                )
-                .emit();
+                );
+                let code = error.obligation.cause.code().peel_derives().peel_match_impls();
+                if let ObligationCauseCode::BindingObligation(..) | ObligationCauseCode::ItemObligation(..) = code {
```

Comment:
This is weird, but I found that other `ObligationCauseCode`s provided strange messages. I can create a diff if the reviewer is curious to see what I mean, but `BindingObligation` and `ItemObligation` are general enough that their message is always pretty useful.